rest/connector-generator/src/main/java/io/syndesis/connector/generator/ConnectorGenerator.java

```diff
@@ -17,35 +17,54 @@
 
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 import io.syndesis.core.Names;
 import io.syndesis.model.connection.Connector;
 import io.syndesis.model.connection.ConnectorTemplate;
+import io.syndesis.model.connection.CustomConnector;
 
-public interface ConnectorGenerator {
+public abstract class ConnectorGenerator {
 
-    default Connector baseConnectorFrom(final ConnectorTemplate connectorTemplate, final Connector template) {
+    public abstract Connector generate(ConnectorTemplate connectorTemplate, CustomConnector customConnector);
```

Comment:
more a question but I always wonder when to use an abstract class vs. an interface with default methods these days. Any hint ?

Reply:
Can't hide implementation details like `baseConnectorFrom` and `determineConnectorName` from clients and I was too lazy to create a base class for generators and keep the interface with only public API.